shader.py

```diff
@@ -19,279 +19,113 @@
-        matprops.gloss = mumat.gloss
-        matprops.specColor = mumat.specColor
-        matprops.shininess = mumat.shininess
-    elif matprops.shader == 'KSP/Alpha/Unlit Transparent':
-        set_tex(mu, matprops.mainTex, mumat.mainTex)
-        matprops.color = mumat.color
-    elif matprops.shader == 'KSP/Unlit':
-        set_tex(mu, matprops.mainTex, mumat.mainTex)
-        matprops.color = mumat.color
-    elif matprops.shader == 'KSP/Diffuse':
-        set_tex(mu, matprops.mainTex, mumat.mainTex)
-    create_nodes(mat)
-    return mat
-
-def shader_update(prop):
-    def updater(self, context):
-        if not hasattr(context, "material"):
-            return
-        mat = context.material
-        if type(self) == MuTextureProperties:
-            pass
-        elif type(self) == MuMaterialProperties:
-            if (prop) == "shader":
-                create_nodes(mat)
-            else:
-                shader = ksp_shaders[mat.mumatprop.shader]
-                nodes = mat.node_tree.nodes
-                for s in shader:
-                    if s[0] == "set" and s[3] == prop:
-                        node_set(mat.name, mat.mumatprop, nodes, s)
-                    elif s[0] == "settex" and s[3] == prop:
-                        node_settex(mat.name, mat.mumatprop, nodes, s)
-    return updater
+    shader = MuEnum.ShaderNames[mumat.type]
+
+    if shader == 'KSP/Specular':
+        set_specular(mumat, mat, mutextures, False)
+    elif shader == 'KSP/Bumped':
+        set_diffuse(mumat, mat, mutextures, False)
+        set_bump(mumat, mat, mutextures)
+    elif shader == 'KSP/Bumped Specular':
+        set_specular(mumat, mat, mutextures, False)
+        set_bump(mumat, mat, mutextures)
+    elif shader == 'KSP/Emissive/Diffuse':
+        set_diffuse(mumat, mat, mutextures, False)
+        set_emissive(mumat, mat, mutextures)
+    elif shader == 'KSP/Emissive/Specular':
+        set_specular(mumat, mat, mutextures, False)
+        set_emissive(mumat, mat, mutextures)
+    elif shader == 'KSP/Emissive/Bumped Specular':
+        set_specular(mumat, mat, mutextures, False)
+        set_emissive(mumat, mat, mutextures)
+        set_bump(mumat, mat, mutextures)
+    elif shader == 'KSP/Alpha/Cutoff':
+        set_diffuse(mumat, mat, mutextures, True)
+    elif shader == 'KSP/Alpha/Cutoff Bumped':
+        set_diffuse(mumat, mat, mutextures, True)
+        set_bump(mumat, mat, mutextures)
+    elif shader == 'KSP/Alpha/Translucent':
+        set_diffuse(mumat, mutextures, True)
+    elif shader == 'KSP/Alpha/Translucent Specular':
+        set_specular(mumat, mutextures, True)
+    elif shader == 'KSP/Alpha/Unlit Transparent':
+        set_diffuse(mumat, mat, mutextures, True)
+        mat.emmit = 1
+    elif shader == 'KSP/Unlit':
+        set_diffuse(mumat, mat, mutextures, False)
+        mat.emmit = 1
+    elif shader == 'KSP/Diffuse':
+        set_diffuse(mumat, mat, mutextures, False)
```

Comment:
FyI could be cleaned using partial functions (e.g. http://chriskiehl.com/article/Cleaner-coding-through-partially-applied-functions/)